feat: Add comprehensive VS Code support for MCP server

harekrishnarai opened the pull request:
## 🎯 **Overview**
This PR adds complete VS Code integration for the pinner-mcp server, making it significantly easier for developers to use the tool in VS Code with GitHub Copilot.

## 📋 **Changes Made**

### ✨ **New Files Added**
- **`.vscode/mcp.json`** - MCP server configuration for VS Code integration
- **`.vscode/settings.json`** - Go development settings with auto-formatting and linting
- **`.vscode/extensions.json`** - Recommended extensions (Go, GitHub Copilot, GitHub Copilot Chat)

### 🔧 **Files Modified**
- **`README.md`** - Added comprehensive VS Code setup instructions with examples
- **`.cursor/mcp.json`** - Fixed Docker image reference to use `ghcr.io/safedep/pinner-mcp:latest`
- **`lefthook.yml`** - Fixed typo: `comands` → `commands`
- **`.gitignore`** - Removed `.vscode/` to include workspace settings in repository

## 🚀 **Features Added**

### VS Code Integration
- **MCP Server Configuration**: Direct integration with GitHub Copilot Chat
- **Developer Experience**: Proper Go tooling setup with auto-formatting and linting
- **Extension Recommendations**: Automatic suggestions for required extensions
- **Team Consistency**: All VS Code settings included in repository

### User Experience Improvements
- **Easy Setup**: Simple copy-paste configuration for VS Code users
- **Clear Documentation**: Step-by-step instructions with examples
- **Consistent Image References**: All configs use the correct `ghcr.io/safedep/pinner-mcp:latest` image

## 💻 **How to Use (VS Code)**

1. **Install Required Extensions** (automatically suggested):
   - Go extension
   - GitHub Copilot
   - GitHub Copilot Chat

2. **MCP Configuration**: The `.vscode/mcp.json` file is already configured

3. **Usage Examples**:
   ```
   Pin GitHub Actions to their commit hash
   ```
   ```
   Pin container base images to digests
   ```
   ```
   Update pinned versions of container base images
   ```


**This PR significantly improves the developer experience for VS Code users while maintaining full backward compatibility with existing workflows.**

harekrishnarai committed:
feat: Add comprehensive VS Code support for MCP server

abhisek commented on .cursor/mcp.json:
```diff
@@ -6,7 +6,7 @@
         "--rm",
         "-i",
-        "safedep/pinner-mcp:latest"
+        "ghcr.io/safedep/pinner-mcp:latest"
```
The local MCP.json file is for local development. When we are developing this project we need a way to build the container and use it locally.

abhisek commented on .vscode/mcp.json:
```diff
@@ -0,0 +1,14 @@
+        "run",
+        "--rm",
+        "-i",
+        "ghcr.io/safedep/pinner-mcp:latest"
```
This config is used only for this project itself right?

Local dev requires testing out unpublished versions

abhisek commented on README.md:
```diff
@@ -16,6 +16,43 @@ Run as a container with `stdio` transport.
+        "run",
+        "--rm",
+        "-i",
+        "ghcr.io/safedep/pinner-mcp:latest"
```
This is fine. Here we are adding docs that is meant for use by other projects.

harekrishnarai committed:
fix: Use local container image for development

Address reviewer feedback by configuring .vscode/mcp.json and .cursor/mcp.json
to use locally built 'pinner-mcp:local' image instead of published image.

This allows developers working on this project to test unpublished changes
during local development, while keeping the published image reference in
the README documentation for end users.

- Updated .vscode/mcp.json to use pinner-mcp:local
- Updated .cursor/mcp.json to use pinner-mcp:local
- Added Development section in README with build instructions

harekrishnarai commented:
## 🔧 Addressed  Feedback

@abhisek Thank you for the review! I've addressed your feedback about local development configuration:

### Changes Made:
1. .vscode/mcp.json - Changed to use pinner-mcp:local for local development
2. .cursor/mcp.json - Changed to use pinner-mcp:local for local development  
3. README.md - Added Development section with build instructions

### Rationale:
- Local Development: The config files in this repo are now for developers working ON this project to test unpublished changes
- End User Documentation: README still shows published image for external users
- Clear Instructions: Added build instructions (docker build -t pinner-mcp:local .)

This maintains the distinction you highlighted - local development configs use local images, while documentation for external users references the published image.
